commands/commands.go

```diff
@@ -141,15 +132,49 @@ func RunWithTimeout(c *Command, fields log.Fields) error {
 	log.Debugf("%s.RunWithTimeout start", c.Name)
 	c.setUpCmd(fields)
 	defer c.closeLogs()
-	log.Debugf("%s.Cmd.Start", c.Name)
+	log.Debugf("%s.Cmd start", c.Name)
+	ctx, cancel := context.WithTimeout(context.Background(), c.TimeoutDuration)
+	defer cancel()
 	if err := c.Cmd.Start(); err != nil {
-		log.Errorf("Unable to start %s: %v", c.Name, err)
+		log.Errorf("unable to start %s: %v", c.Name, err)
+		return err
+	}
+	go func() {
+		select {
+		case <-ctx.Done():
+			// if the context was canceled we don't want to kill the
+			// process because it's already gone
+			if ctx.Err().Error() == "context deadline exceeded" {
```

Comment:
Derp, just saw that I missed the `const` in the `context` package for this. Well in _most_ of the stdlib they're lacking errors... I'll fix this by checking vs the `DeadlineExceeded` error.